src/datahike/writing.cljc

```diff
@@ -0,0 +1,239 @@
+(ns datahike.writing
+  "Manage all state changes and access to state of durable store."
+  (:require [datahike.db :as db]
+            [datahike.db.utils :as dbu]
+            [datahike.index :as di]
+            [datahike.store :as ds]
+            [datahike.tools :as dt]
+            [datahike.core :as core]
+            [datahike.config :as dc]
+            [konserve.core :as k]
+            [taoensso.timbre :as log]
+            [hasch.core :refer [uuid]]))
+
+;; mapping to storage
+
+(defn stored-db? [obj]
+  ;; TODO use proper schema to match?
+  (let [keys-to-check [:eavt-key :aevt-key :avet-key :schema :rschema
```

Comment:
> Looks well written in total but it lacks testing. With such a large new use case we should have a decent amount of tests covering the new feature.

I agree, but note that the code does not change the way the connection works for existing use cases and hence is tested for them with the current code base. Accessing a remote connection should be tested with an integration test including `datahike-server` and `datahike-server-transactor`. Since we also want to show case that, it would be good for me to get some feedback how to do it. We can then add an integration test in the same way we do it for datahike with konserve. Is that ok? I still think that this can be merged now then and the additional test can be added with the new use case.